database/src/migrations/20230830143800_create_survey_site_strategies.ts

```diff
@@ -0,0 +1,244 @@
+
+/**
+ * 1. Adds two new tables:
+ *  - Codes table for site selection strategy 
+ *  - Xref table for survey site selection strategies
+ * 2. Updates the survey delete procedure to account for blocks, stratums and survey site selection strategies
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+
+  ----------------------------------------------------------------------------------------
+  -- Create new site_strategy table
+  ----------------------------------------------------------------------------------------
+
+  SET search_path=biohub;
+
+  CREATE TABLE site_strategy(
+    site_strategy_id                        integer           GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+    name                                    varchar(50)       NOT NULL,
+    description                             varchar(250),
+    record_effective_date                   date              NOT NULL,
+    record_end_date                         date,
+    create_date                             timestamptz(6)    DEFAULT now() NOT NULL,
+    create_user                             integer           NOT NULL,
+    update_date                             timestamptz(6),
+    update_user                             integer,
+    revision_count                          integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT site_strategy_pk PRIMARY KEY (site_strategy_id)
+  );
+
+
+  COMMENT ON COLUMN site_strategy.site_strategy_id IS 'System generated surrogate primary key identifier.'
+  ;
+  COMMENT ON COLUMN site_strategy.name IS 'The name of the site selection strategy.'
+  ;
+  COMMENT ON COLUMN site_strategy.record_effective_date IS 'Record level effective date.'
+  ;
+  COMMENT ON COLUMN site_strategy.description IS 'The description of the site selection strategy.'
+  ;
+  COMMENT ON COLUMN site_strategy.record_end_date IS 'Record level end date.'
+  ;
+  COMMENT ON COLUMN site_strategy.create_date IS 'The datetime the record was created.'
+  ;
+  COMMENT ON COLUMN site_strategy.create_user IS 'The id of the user who created the record as identified in the system user table.'
+  ;
+  COMMENT ON COLUMN site_strategy.update_date IS 'The datetime the record was updated.'
+  ;
+  COMMENT ON COLUMN site_strategy.update_user IS 'The id of the user who updated the record as identified in the system user table.'
+  ;
+  COMMENT ON COLUMN site_strategy.revision_count IS 'Revision count used for concurrency control.'
+  ;
+  COMMENT ON TABLE site_strategy IS 'Broad classification for the site_strategy code of the survey.'
+  ;
+
+
+  ----------------------------------------------------------------------------------------
+  -- Create new keys and indices
+  ----------------------------------------------------------------------------------------
+
+  -- Add unique constraint
+  CREATE UNIQUE INDEX site_strategy_uk1 ON site_strategy(name, record_end_date);
```

Comment:
double check the other examples of this "nuk1", there are a few other things needed to make this work, as a result of how Postgres handles NULL values (or see the database readme, I put an explanation in there about it, as its super not obvious).